### PR TITLE
Align transaction handling

### PR DESCRIPTION
**Pull request by albertlast**
Drop the transaction handling in pg because mysql doesn't support this.
And i'm lazy -> i build this pr on the rewrite db_insert_id pr
to see only the real changes: https://github.com/albertlast/SMF2.1/commit/df30aa0f3a5118f0f256569e03e2e1697837be65

### CODE REVIEW

**Comment by jdarwood007** [comment]
MySQL supports transactions http://dev.mysql.com/doc/refman/5.7/en/commit.html

I'm having trouble finding when it was added, but far as I can tell it has existed since 3.23.28 and is only supported on InnoDB tables (no MyISAM).

**Comment by albertlast** [comment]
i know that it's support transaction,
but i don't like that db_insert works differently between mysql and pg.
the other way would be to add this to mysql too.

**Comment by jdarwood007** [comment]
Ok, I see your commit now.  Removing it from the insert function is fine, if we need to call the transaction we can do it ourselves rather than in the insert function.

File: Sources/RepairBoards.php [hunk 1]
```diff
@@ -402,12 +401,11 @@ function fix_full_processing:
 							'xx',
 							1,
 						),
-						array('id_topic')
+						array('id_topic'),
```
Comment (live627) [inline Sources/RepairBoards.php]: This should be `id_msg`

File: Sources/RepairBoards.php [hunk 2]
```diff
@@ -538,12 +535,11 @@ function fix_full_processing:
 						'xx',
 						1,
 					),
-					array('id_topic')
+					array('id_topic'),
```
Comment (live627) [inline Sources/RepairBoards.php]: Same as above.

**Review by live627 — CHANGES_REQUESTED** [review]

**Pull request merged**